core/src/main/java/org/apache/calcite/runtime/CalciteResource.java

```diff
@@ -112,6 +112,9 @@ public interface CalciteResource {
   @BaseMessage("Illegal identifier '':''. Was expecting ''VALUE''")
   ExInst<CalciteException> illegalColon();
 
+  @BaseMessage("Illegal identifier '',''. Was expecting ''VALUE''")
```

Comment:
Comma is not really an identifier.  Maybe "symbol".
Yes, the previous message is probably wrong too.

Reply:
It's kind of a delimiter too? Wdyt

Reply:
delimiter is also fine

Reply:
Well `,` isn't a legal identifier - the message is not wrong. It's kind of a two-prong message; we were expecting an identifier, and for the identifier to be `VALUE`. We fell at the first hurdle because `,` isn't an identifier.

(AFAICT `VALUE` is usually a keyword not an identifier but it's a bit squishy because it could easily be a non-reserved word.)

A better message might be "Unexpected symbol" rather than "Illegal identifier", which has better symmetry with the expected clause following. I'll change both to that.